### ISSUE TITLE
"Invalid proto descriptor for file "tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto"

### ISSUE THREAD

**Issue by yaroslavvb** [tensorflow/tensorflow#6199]
[trace_info.proto](https://github.com/tensorflow/tensorflow/blob/287db3a9b0701021f302e7bb58af5cf89fdcd424/tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto) seems to be missing `import tensorflow/core/protobuf/meta_graph.proto`. It works for Python protobuf implementation but fails with tracing runs on stricter cpp implementation:

[scratch.py](https://gist.github.com/yaroslavvb/b0b9e51d88bb97aa0f739f8130e1c3d5)

```
python scratch.py

...
    contrib = importlib.import_module('tensorflow.contrib')
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 986, in _gcd_import
  File "<frozen importlib._bootstrap>", line 969, in _find_and_load
  File "<frozen importlib._bootstrap>", line 958, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 673, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 665, in exec_module
  File "<frozen importlib._bootstrap>", line 222, in _call_with_frames_removed
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/__init__.py", line 49, in <module>
    from tensorflow.contrib import tensorboard
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/tensorboard/__init__.py", line 22, in <module>
    from tensorflow.contrib.tensorboard import plugins
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/tensorboard/plugins/__init__.py", line 23, in <module>
    from tensorflow.contrib.tensorboard.plugins import trace
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/tensorboard/plugins/trace/__init__.py", line 22, in <module>
    from tensorflow.contrib.tensorboard.plugins.trace.trace import *
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/tensorboard/plugins/trace/trace.py", line 28, in <module>
    from tensorflow.contrib.tensorboard.plugins.trace.trace_info_pb2 import TraceInfo
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/tensorflow/contrib/tensorboard/plugins/trace/trace_info_pb2.py", line 22, in <module>
    serialized_pb=_b('\n=tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto\x12\ntensorflow\"Q\n\tTraceInfo\x12\x1f\n\x03ops\x18\x01 \x03(\x0b\x32\x12.tensorflow.OpInfo\x12#\n\x05\x66iles\x18\x02 \x03(\x0b\x32\x14.tensorflow.FileInfo\"\xb2\x01\n\x06OpInfo\x12\x0c\n\x04name\x18\x01 \x01(\t\x12\x0f\n\x07op_type\x18\x02 \x01(\t\x12\x0e\n\x06\x64\x65vice\x18\x03 \x01(\t\x12(\n\ttraceback\x18\x04 \x03(\x0b\x32\x15.tensorflow.LineTrace\x12&\n\x06inputs\x18\x05 \x03(\x0b\x32\x16.tensorflow.TensorInfo\x12\'\n\x07outputs\x18\x06 \x03(\x0b\x32\x16.tensorflow.TensorInfo\"3\n\tLineTrace\x12\x11\n\tfile_path\x18\x01 \x01(\t\x12\x13\n\x0bline_number\x18\x02 \x01(\r\"Y\n\nTensorInfo\x12\r\n\x05shape\x18\x01 \x03(\x05\x12\r\n\x05\x64type\x18\x02 \x01(\t\x12\x1a\n\x12num_bytes_per_elem\x18\x03 \x01(\r\x12\x11\n\tconsumers\x18\x04 \x03(\t\"\xbb\x01\n\x08\x46ileInfo\x12\x11\n\tfile_path\x18\x01 \x01(\t\x12\x13\n\x0bsource_code\x18\x02 \x01(\t\x12K\n\x14multiline_statements\x18\x03 \x03(\x0b\x32-.tensorflow.FileInfo.MultilineStatementsEntry\x1a:\n\x18MultilineStatementsEntry\x12\x0b\n\x03key\x18\x01 \x01(\r\x12\r\n\x05value\x18\x02 \x01(\r:\x02\x38\x01\x62\x06proto3')
  File "/home/yaroslav/.conda/envs/openai/lib/python3.5/site-packages/google/protobuf/descriptor.py", line 827, in __new__
    return _message.default_pool.AddSerializedFile(serialized_pb)
TypeError: Couldn't build proto file into descriptor pool!
Invalid proto descriptor for file "tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto":
  tensorflow.TensorInfo.dtype: "tensorflow.TensorInfo.dtype" is already defined in file "tensorflow/core/protobuf/meta_graph.proto".
  tensorflow.TensorInfo: "tensorflow.TensorInfo" is already defined in file "tensorflow/core/protobuf/meta_graph.proto".
  tensorflow.OpInfo.inputs: "tensorflow.TensorInfo" seems to be defined in "tensorflow/core/protobuf/meta_graph.proto", which is not imported by "tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto".  To use it here, please add the necessary import.
  tensorflow.OpInfo.outputs: "tensorflow.TensorInfo" seems to be defined in "tensorflow/core/protobuf/meta_graph.proto", which is not imported by "tensorflow/contrib/tensorboard/plugins/trace/trace_info.proto".  To use it here, please add the necessary import.

```

`__git_version__: "b'0.12.0-rc0-437-g6d63f67'"` ( pulled from head Dec 6th 5pm)

**Comment by dsmilkov** [comment]
Strange that you are seeing this error at HEAD. This was due to both `meta_graph.proto` and `trace_info.proto` having the same package `tensorflow` and defining the same message `TensorInfo`.
This got fixed 10 days ago with commit efd47f2418629245740f9c7d775162cc80d197fc

Did you install tensorflow from head or did you use the 0.12 RC0 version? Let me know if the error still persists at head. Thanks!

**Comment by yaroslavvb** [comment]
Aha, it seems I don't have that change. While your commit entered internal branch on Dec 2, it was [merged](https://github.com/tensorflow/tensorflow/commit/e624a5f4) into HEAD on Dec 6 at 8:10 PM, so I was using pre-patch version. I'll just assume this works now and re-open if it doesn't
